### PR TITLE
8304693: Remove -XX:-UseVtableBasedCHA

### PR DESCRIPTION
**Pull request by iwanowww**
JDK-8266074 introduced new CHA implementation and a flag (-XX:-UseVtableBasedCHA) to switch back to the original implementation for diagnostic purposes. Vtable-based CHA implementation has been turned on by default since 17 and now the time has come to remove UseVtableBasedCHA flag.

Testing: hs-tier1 - hs-tier6

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8304693](https://bugs.openjdk.org/browse/JDK-8304693): Remove -XX:-UseVtableBasedCHA (**Enhancement** - P4)

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19911/head:pull/19911` \
`$ git checkout pull/19911`

Update a local copy of the PR: \
`$ git checkout pull/19911` \
`$ git pull https://git.openjdk.org/jdk.git pull/19911/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19911`

View PR using the GUI difftool: \
`$ git pr show -t 19911`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19911.diff">https://git.openjdk.org/jdk/pull/19911.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19911#issuecomment-2192309219)

### CODE REVIEW

File: src/hotspot/share/runtime/arguments.cpp [hunk 1]
```diff
@@ -522,6 +522,7 @@ static SpecialFlag const special_jvm_flags[] = {
 #endif // X86
 
   { "HeapFirstMaximumCompactionCount", JDK_Version::undefined(), JDK_Version::jdk(24), JDK_Version::jdk(25) },
+  { "UseVtableBasedCHA",               JDK_Version::undefined(), JDK_Version::jdk(24), JDK_Version::jdk(25) },
```
Comment (coleenp) [inline src/hotspot/share/runtime/arguments.cpp:525]: We don't usually put diagnostic VM options in this table.

**Review by vnkozlov — APPROVED** [review]
Good.

**Review by coleenp — COMMENTED** [review]
Thanks for fixing the add_to_hierarchy code.

File: src/hotspot/share/runtime/arguments.cpp (hunk 1 above)
Comment (iwanowww) [reply src/hotspot/share/runtime/arguments.cpp:525]: Ok, I was confused by the following comment before the table: 
```
 * To remove internal options (e.g. diagnostic, experimental, develop options), use
 * a 2-step model adding major release numbers to the obsolete and expire columns.
```
Does it make sense to adjust it as well?

Comment (coleenp) [reply src/hotspot/share/runtime/arguments.cpp:525]: Oh, right, that is what the comment says. I'd forgotten about the 2-step process.

**Review by coleenp — APPROVED** [review]
Looks good!

**Comment by iwanowww** [comment]
Thanks for the reviews, Vladimir, Coleen, and David.

/integrate

**Pull request closed without merging**